MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -1,72 +1,178 @@
+        
+        imageView.layer.borderWidth = 0
+        
+        textLabel.text = question.question
+        counterLabel.text = question.questionNumber
+        imageView.image = question.image
+    }
+    
+    private func showResult() {
+        quizCount += 1
+        stats.append(Statistic(result: correctAnswers, date: Date()))
+        
+        let record: Statistic? = stats.max { a, b in a.result < b.result }
+        let sumResults: Int = stats.reduce(0) { $0 + $1.result }
+        let totalQuestionsCount: Int = quizCount * questions.count
+        let averagePercentage: Float = Float(sumResults) / Float(totalQuestionsCount) * 100
+        var message = ""
+
+        if (record != nil) {
+            let formatter = DateFormatter()
+            formatter.dateFormat = "dd.MM.yyyy HH:mm"
+            let date = formatter.string(from: record!.date)
+            
+            let resultText = "Ваш результат: \(self.correctAnswers)/\(self.questions.count)"
+            let quizCountText = "Количество сыгранных квизов: \(self.quizCount)"
+            let recordText = "Рекорд: \(record!.result)/\(self.questions.count) (\(date))"
+            let averageText = "Средняя точность: \(String(format: "%.2f", averagePercentage))%"
+            
+            message = "\(resultText)\n\(quizCountText)\n\(recordText)\n\(averageText)"
+        }
+        
+        let alert = UIAlertController(title: "Этот раунд окончен! ", message: message, preferredStyle: .alert)
+        
+        let action = UIAlertAction(title: "Сыграть ещё раз", style: .default, handler: { _ in
+            self.correctAnswers = 0
+            self.currentQuestionIndex = 0
+            self.show(quiz: self.questions[self.currentQuestionIndex])
+        })
+        
+        alert.addAction(action)
+        
+        self.present(alert, animated: true, completion: nil)
+    }
+    
+    private func convert(model: QuizQuestion) -> QuizStepViewModel {
+        return QuizStepViewModel(
+            image: UIImage(named: model.image) ?? UIImage(),
+            question: model.text,
+            questionNumber: "\(currentQuestionIndex + 1)/\(questions.count)"
+        )
+    }
+    
+    private func showAnswerResult(isCorrect: Bool) {
+        if (isCorrect) {
+            correctAnswers += 1
+        }
+        
+        noButton.isEnabled = false
+        yesButton.isEnabled = false
+
+        imageView.layer.masksToBounds = true
+        imageView.layer.borderWidth = 8
+        imageView.layer.borderColor = isCorrect ? UIColor.ypGreen.cgColor : UIColor.ypRed.cgColor
+        imageView.layer.cornerRadius = 20
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
+            self.showNextQuestionOrResults()
+            
+            self.noButton.isEnabled = true
+            self.yesButton.isEnabled = true
+        }
+    }
+    
+    private func showNextQuestionOrResults() {
+        if (currentQuestionIndex == questions.count - 1) {
```

Comment:
в swift скобки избыточны, их используют лишь для группировки более длинных логических выражений для читаемости
```suggestion
        if currentQuestionIndex == questions.count - 1 {
```